# Profitable App Profiles for the App Store and Google Play Markets

Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets. We're working as data analysts for a company that builds Android and iOS mobile apps, and our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

## Opening and Exploring the Data
As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

![img](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png) Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)

Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of data instead. To avoid spending resources with collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our purpose:

- [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
- [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

Let's start by opening the two data sets and then continue with exploring the data.

In [916]:
import pandas as pd
ios = pd.read_csv(r"C:\Users\ntxhi\Downloads\AppleStore.csv")
android = pd.read_csv(r"C:\Users\ntxhi\Downloads\googleplaystore.csv")

### Analyze App Store Dataset

In [917]:
ios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                7197 non-null   int64  
 1   track_name        7197 non-null   object 
 2   size_bytes        7197 non-null   int64  
 3   currency          7197 non-null   object 
 4   price             7197 non-null   float64
 5   rating_count_tot  7197 non-null   int64  
 6   rating_count_ver  7197 non-null   int64  
 7   user_rating       7197 non-null   float64
 8   user_rating_ver   7197 non-null   float64
 9   ver               7197 non-null   object 
 10  cont_rating       7197 non-null   object 
 11  prime_genre       7197 non-null   object 
 12  sup_devices.num   7197 non-null   int64  
 13  ipadSc_urls.num   7197 non-null   int64  
 14  lang.num          7197 non-null   int64  
 15  vpp_lic           7197 non-null   int64  
dtypes: float64(3), int64(8), object(5)
memory 

In [918]:
ios.describe()

,id,size_bytes,price,rating_count_tot,rating_count_ver,user_rating,user_rating_ver,sup_devices.num,ipadSc_urls.num,lang.num,vpp_lic
count,7.197000e+03,7.197000e+03,7197.000000,7.197000e+03,7197.000000,7197.000000,7197.000000,7197.000000,7197.000000,7197.000000,7197.000000
mean,8.631310e+08,1.991345e+08,1.726218,1.289291e+04,460.373906,3.526956,3.253578,37.361817,3.707100,5.434903,0.993053
std,2.712368e+08,3.592069e+08,5.833006,7.573941e+04,3920.455183,1.517948,1.809363,3.737715,1.986005,7.919593,0.083066
min,2.816565e+08,5.898240e+05,0.000000,0.000000e+00,0.000000,0.000000,0.000000,9.000000,0.000000,0.000000,0.000000
25%,6.000937e+08,4.692275e+07,0.000000,2.800000e+01,1.000000,3.500000,2.500000,37.000000,3.000000,1.000000,1.000000
50%,9.781482e+08,9.715302e+07,0.000000,3.000000e+02,23.000000,4.000000,4.000000,37.000000,5.000000,1.000000,1.000000
75%,1.082310e+09,1.819249e+08,1.990000,2.793000e+03,140.000000,4.500000,4.500000,38.000000,5.000000,8.000000,1.000000
max,1.188376e+09,4.025970e+09,299.990000,2.974676e+06,177050.000000,5.000000,5.000000,47.000000,5.000000,75.000000,1.000000


In [919]:
ios.head()

,id,track_name,size_bytes,currency,price,rating_count_tot,rating_count_ver,user_rating,user_rating_ver,ver,cont_rating,prime_genre,sup_devices.num,ipadSc_urls.num,lang.num,vpp_lic
0,284882215,Facebook,389879808,USD,0.0,2974676,212,3.5,3.5,95.0,4+,Social Networking,37,1,29,1
1,389801252,Instagram,113954816,USD,0.0,2161558,1289,4.5,4.0,10.23,12+,Photo & Video,37,0,29,1
2,529479190,Clash of Clans,116476928,USD,0.0,2130805,579,4.5,4.5,9.24.12,9+,Games,38,5,18,1
3,420009108,Temple Run,65921024,USD,0.0,1724546,3842,4.5,4.0,1.6.2,9+,Games,40,5,1,1
4,284035177,Pandora - Music & Radio,130242560,USD,0.0,1126879,3594,4.0,4.5,8.4.1,12+,Music,37,4,1,1


The App Store Dataset has **7197 rows and 16 column**. The columns that could help us with our analysis are:

> track_name, currency, price, rating_count_tot, user_rating, prime_genre

### Analyze Google Play Dataset

In [920]:
android.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [921]:
android.describe()

,Rating
count,9367.000000
mean,4.193338
std,0.537431
min,1.000000
25%,4.000000
50%,4.300000
75%,4.500000
max,19.000000


In [922]:
android.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [923]:
android.query("App == 'Coloring book moana'")

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2033,Coloring book moana,FAMILY,3.9,974,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up


  
The Google Play dataset has **10841 rows and 13 column**. The columns that could help us with our analysis are:

> App, Category, Reviews, Installs, Type, Price, Genres

## Cleaning Data

### Deleting Wrong Data

In the previous step, we opened the two data sets and explored the data. Before beginning our analysis, we need to make sure the data we analyze is accurate, or the results of our analysis will be wrong. This means that we need to do the following:

- Detect inaccurate data, and correct or remove it.
- Detect duplicate data, and remove the duplicates.

Recall that at our company, we only build apps that are _free_ to download and install, and we design them for an _English-speaking_ audience. This means that we'll need to do the following:

- Remove non-English apps like _爱奇艺PPS -《欢乐颂2》电视剧热播_.
- Remove apps that aren't free.

We call this process of preparing our data for analysis **data cleaning**. We do data cleaning _before_ the analysis; it includes removing or correcting wrong data, removing duplicate data, and modifying the data to fit the purpose of our analysis.

It's often said that data scientists spend around 80% of their time cleaning data, and only about 20% actually analyzing (cleaned) data. In this project, we'll see that this is not far from the truth.

Let's begin by detecting and deleting wrong data. For this guided project, we'll guide you through the entire data cleaning process. In later courses, we'll learn more about data cleaning, and you'll be able to perform data cleaning without any guidance.

If you get stuck during the following exercise, you can check the [solution notebook](https://github.com/dataquestio/solutions/blob/master/Mission350Solutions.ipynb).

The Google Play data set has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), and we can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) outlines an error for row 10472. Let's print this row and compare it against the header and another row that is correct.

In [924]:
# The Google Play dataset 
android.iloc[10472]

App               Life Made WI-Fi Touchscreen Photo Frame
Category                                              1.9
Rating                                               19.0
Reviews                                              3.0M
Size                                               1,000+
Installs                                             Free
Type                                                    0
Price                                            Everyone
Content Rating                                        NaN
Genres                                  February 11, 2018
Last Updated                                       1.0.19
Current Ver                                    4.0 and up
Android Ver                                           NaN
Name: 10472, dtype: object

The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that _the rating is NaN_. This is clearly off because **the maximum rating for a Google Play app is 5** (as mentioned in the discussions section, this problem is caused by a missing value in the 'Category' column). As a consequence, we'll delete this row.

In [925]:
android = android.drop(10472)

In [926]:
android.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10840 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10840 non-null  object 
 1   Category        10840 non-null  object 
 2   Rating          9366 non-null   float64
 3   Reviews         10840 non-null  object 
 4   Size            10840 non-null  object 
 5   Installs        10840 non-null  object 
 6   Type            10839 non-null  object 
 7   Price           10840 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10840 non-null  object 
 10  Last Updated    10840 non-null  object 
 11  Current Ver     10832 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.2+ MB


### Removing Duplicate Entries[](https://jupyter-hub.dataquest.io/user/app-m-350-u-1627797/lab/workspaces/auto-r/tree/Basics.ipynb#Removing-Duplicate-Entries)

#### Part One[](https://jupyter-hub.dataquest.io/user/app-m-350-u-1627797/lab/workspaces/auto-r/tree/Basics.ipynb#Part-One)

In the previous step, we started the data cleaning process and deleted a row with incorrect data from the Google Play dataset. If you explore the Google Play data set long enough or look at the discussions section, you'll notice some apps have duplicate entries. For instance, Instagram has four entries:

##### The Google Play dataset[¶](https://jupyter-hub.dataquest.io/user/app-m-350-u-1627797/lab/workspaces/auto-r/tree/Basics.ipynb#The-Google-Play-dataset)

In [927]:
android.query("App == 'Instagram'")

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
2545,Instagram,SOCIAL,4.5,66577313,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
2604,Instagram,SOCIAL,4.5,66577446,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
2611,Instagram,SOCIAL,4.5,66577313,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
3909,Instagram,SOCIAL,4.5,66509917,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device


In total, there are 1,181 cases where an app occurs more than once:

In [928]:
len(android) - android['App'].nunique()

1181

#### Part Two

We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows we printed for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show the data was collected at different times.

![img](https://s3.amazonaws.com/dq-content/350/py1m8_fourth_col.png)

We can use this information to build a criterion for removing the duplicates. The higher the number of reviews, the more recent the data should be. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries for any given app.

In [929]:
android.sort_values(['App', 'Reviews'], ascending=[True, False]).query("App == 'Instagram'")

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
2604,Instagram,SOCIAL,4.5,66577446,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
2545,Instagram,SOCIAL,4.5,66577313,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
2611,Instagram,SOCIAL,4.5,66577313,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device
3909,Instagram,SOCIAL,4.5,66509917,Varies with device,"1,000,000,000+",Free,0,Teen,Social,"July 31, 2018",Varies with device,Varies with device


In [930]:
# Sort dataset by appplication names and total of reviews in ascending and descending order respectively to remain only the application with the highest number of reviews
android = android.sort_values(['App', 'Reviews'], ascending=[True, False]) \
                 .drop_duplicates(subset=['App']) \
                 .sort_index() 

In [931]:
android

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up
5,Paper flowers instructions,ART_AND_DESIGN,4.4,167,5.6M,"50,000+",Free,0,Everyone,Art & Design,"March 26, 2017",1.0,2.3 and up
...,...,...,...,...,...,...,...,...,...,...,...,...,...
10836,Sya9a Maroc - FR,FAMILY,4.5,38,53M,"5,000+",Free,0,Everyone,Education,"July 25, 2017",1.48,4.1 and up
10837,Fr. Mike Schmitz Audio Teachings,FAMILY,5.0,4,3.6M,100+,Free,0,Everyone,Education,"July 6, 2018",1.0,4.1 and up
10838,Parkinson Exercices FR,MEDICAL,NaN,3,9.5M,"1,000+",Free,0,Everyone,Medical,"January 20, 2017",1.0,2.2 and up
10839,The SCP Foundation DB fr nn5n,BOOKS_AND_REFERENCE,4.5,114,Varies with device,"1,000+",Free,0,Mature 17+,Books & Reference,"January 19, 2015",Varies with device,Varies with device


We don't need to do the same for the App Store data because there are no duplicates.

In [932]:
ios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                7197 non-null   int64  
 1   track_name        7197 non-null   object 
 2   size_bytes        7197 non-null   int64  
 3   currency          7197 non-null   object 
 4   price             7197 non-null   float64
 5   rating_count_tot  7197 non-null   int64  
 6   rating_count_ver  7197 non-null   int64  
 7   user_rating       7197 non-null   float64
 8   user_rating_ver   7197 non-null   float64
 9   ver               7197 non-null   object 
 10  cont_rating       7197 non-null   object 
 11  prime_genre       7197 non-null   object 
 12  sup_devices.num   7197 non-null   int64  
 13  ipadSc_urls.num   7197 non-null   int64  
 14  lang.num          7197 non-null   int64  
 15  vpp_lic           7197 non-null   int64  
dtypes: float64(3), int64(8), object(5)
memory 

### [Removing Non-English Apps:](https://jupyter-hub.dataquest.io/user/app-m-350-u-1627797/lab/workspaces/auto-r/tree/Basics.ipynb#Removing-Non-English-Apps:)

#### Part One

In the previous step, we managed to remove the duplicate app entries in the Google Play dataset. We don't need to do the same for the App Store data because there are no duplicates.

Remember we use English for the apps we develop at our company, and we'd like to analyze only the apps that are designed for an English-speaking audience. However, if we explore the data long enough, we'll find that both datasets have apps with names that suggest they are not designed for an English-speaking audience.

In [933]:
ios.iloc[813, 1]

'爱奇艺PPS -《欢乐颂2》电视剧热播'

In [934]:
ios.iloc[6731, 1]

'【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜'

In [935]:
# android.loc[android.query("App == '中国語 AQリスニング'").index[0], 'App']
android.query("App == '中国語 AQリスニング'").index[0]
# android.iloc[4415, 0]

np.int64(5513)

In [936]:
# android.loc[android.query("App == 'لعبة تقدر تربح DZ'").index[0], 'App']
# android.query("App == 'لعبة تقدر تربح DZ'").index[0]
android.iloc[7941, 0]


'لعبة تقدر تربح DZ'

![img](https://s3.amazonaws.com/dq-content/350/py1m8_non_english.png)

We're not interested in keeping these apps, so we'll remove them. One way to do this is to remove each app with a name containing a symbol that isn't commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /).

Each character we use in a string has a corresponding number associated with it. For instance, the corresponding number for character 'a' is 97, character 'A' is 65, and character '爱' is 29,233. We can get the corresponding number of each character using the [ord() built-in function](https://docs.python.org/3/library/functions.html#ord).

![img](https://s3.amazonaws.com/dq-content/350/py1m8_ord.png)

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the [ASCII](https://en.wikipedia.org/wiki/ASCII) (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

If an app name contains a character that is greater than 127, then it probably means that the app has a non-English name. Our app names, however, are stored as strings, so how could we take each individual character of a string and check its corresponding number?

In Python, strings are indexable and iterable, which means we can use indexing to select an individual character, and we can also iterate on the string using a for loop.

![img](https://s3.amazonaws.com/dq-content/350/py1m8_str.png)

#### Part Two

In [937]:
# Before removing non-English apps
ios.shape

(7197, 16)

In [938]:
# Filter rows with English apps
ios_temp = ios.loc[ios['track_name'].apply(lambda x: str(x).isascii()) == True]
ios_temp.shape

(5707, 16)

In [939]:
# Before removing non-English apps
android.shape

(9659, 13)

In [940]:
# Filter rows with English named apps
android_temp = android.loc[android['App'].apply(lambda x: str(x).isascii()) == True]
android_temp.shape

(9117, 13)

However, we notice that some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, we'll remove useful apps if we use the above approach.

In [941]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


#### Better approach

Use a function that:

1. keeps names with English letters
2. rejects names containing non-Latin scripts like Chinese, Arabic, Cyrillic, Japanese

Example:

In [942]:
import re

non_english_script = re.compile(
    r'[\u0400-\u04FF]'   # Cyrillic
    r'|[\u0600-\u06FF]'  # Arabic
    r'|[\u0590-\u05FF]'  # Hebrew
    r'|[\u4E00-\u9FFF]'  # CJK Unified Ideographs
    r'|[\u3040-\u30FF]'  # Hiragana/Katakana
    r'|[\u1100-\u11FF]'  # Hangul Jamo
)

def is_english_app(name):
    name = str(name)
    return bool(re.search(r'[A-Za-z]', name)) and not non_english_script.search(name)

android = android.loc[android['App'].apply(is_english_app)]
ios = ios.loc[ios['track_name'].apply(is_english_app)]


In [943]:
android.shape

(9619, 13)

In [944]:
ios.shape

(6137, 16)

We can see that we're left with 9619 Android apps and 6137 iOS apps.

### Isolating the Free Apps

So far in the data cleaning process, we've done the following:

- Removed inaccurate data
- Removed duplicate app entries
- Removed non-English apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our datasets contain both free and non-free apps; we'll need to isolate only the free apps for our analysis.

Isolating the free apps will be our last step in the data cleaning process. On the next screen, we're going to start analyzing the data.



In [945]:
ios = ios.query("price == 0")

In [946]:
ios.shape

(3183, 16)

In [947]:
android = android.query("Type == 'Free'")

In [948]:
android.shape

(8867, 13)

We're left with 8867 Android apps and 3183 iOS apps, which should be enough for our analysis.

### Most Common Apps by Genre

So far, we've spent a good amount of time cleaning data, including the following:

- Removing inaccurate data
- Removing duplicate app entries
- Removing non-English apps
- Isolating the free apps

As we mentioned in the introduction, our goal is to determine the kinds of apps that are likely to attract more users because the number of people using our apps affect our revenue.

To minimize risks and overhead, our validation strategy for an app idea has three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful in both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by determining the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our datasets.

We start by examining the frequency table for the `prime_genre` column of the App Store data set.

In [949]:
ios['prime_genre'] \
    .value_counts(normalize=True) \
    .mul(100) \
    .round(2)

prime_genre
Games                58.31
Entertainment         7.92
Photo & Video         5.03
Education             3.68
Social Networking     3.27
Shopping              2.54
Utilities             2.42
Sports                2.17
Music                 2.04
Health & Fitness      2.04
Productivity          1.73
Lifestyle             1.57
News                  1.35
Travel                1.19
Finance               1.10
Weather               0.88
Food & Drink          0.82
Reference             0.53
Business              0.53
Book                  0.38
Navigation            0.19
Medical               0.19
Catalogs              0.13
Name: proportion, dtype: float64

We can see that among the free English apps, more than a half (**58.31%**) are `Games`. `Entertainment` apps are close to **8%**, followed by `Photo & Video` apps, which are around **5%**. Only **3.68%** of the apps are designed for `Education`, followed by `Social Networking` apps which amount for **3.27%** of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Let's continue by examining the `Genres` and `Category` columns of the Google Play data set (two columns which seem to be related).

In [950]:
android['Genres'] \
    .value_counts(normalize=True) \
    .mul(100) \
    .round(2)

Genres
Tools                       8.42
Entertainment               6.10
Education                   5.35
Business                    4.60
Lifestyle                   3.91
                            ... 
Role Playing;Brain Games    0.01
Strategy;Education          0.01
Racing;Pretend Play         0.01
Communication;Creativity    0.01
Strategy;Creativity         0.01
Name: proportion, Length: 115, dtype: float64

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

[![img](https://camo.githubusercontent.com/ccb05d240421d1c8fc78daad94898d8097ebeab445e98a96f34ac99e4bc68a26/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f64712d636f6e74656e742f3335302f7079316d385f66616d696c792e706e67)](https://play.google.com/store/apps/category/FAMILY?hl=en)

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the `Genres` column:

In [951]:
android['Category'] \
    .value_counts(normalize=True) \
    .mul(100) \
    .round(2)

Category
FAMILY                 18.95
GAME                    9.68
TOOLS                   8.44
BUSINESS                4.60
LIFESTYLE               3.92
PRODUCTIVITY            3.90
FINANCE                 3.70
MEDICAL                 3.52
SPORTS                  3.38
PERSONALIZATION         3.33
COMMUNICATION           3.24
HEALTH_AND_FITNESS      3.08
PHOTOGRAPHY             2.94
NEWS_AND_MAGAZINES      2.80
SOCIAL                  2.66
TRAVEL_AND_LOCAL        2.33
SHOPPING                2.24
BOOKS_AND_REFERENCE     2.15
DATING                  1.86
VIDEO_PLAYERS           1.80
MAPS_AND_NAVIGATION     1.40
FOOD_AND_DRINK          1.24
EDUCATION               1.17
ENTERTAINMENT           0.96
LIBRARIES_AND_DEMO      0.94
AUTO_AND_VEHICLES       0.92
HOUSE_AND_HOME          0.81
WEATHER                 0.80
EVENTS                  0.71
PARENTING               0.65
ART_AND_DESIGN          0.64
COMICS                  0.62
BEAUTY                  0.60
Name: proportion, dtype: float64

The difference between the `Genres` and the `Category` columns is not crystal clear, but one thing we can notice is that the `Genres` column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the `Category` column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

### Most Popular Apps by Genre on the App Store

The frequency tables we analyzed on the previous screen showed us that apps designed for fun dominate the App Store, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we'd like to determine the kind of apps with the most users.

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` column.

Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to do the following:

- Isolate the apps of each genre
- Add up the user ratings for the apps of that genre
- Divide the sum by the number of apps belonging to that genre (not by the total number of apps)



In [952]:
ios.pivot_table(index='prime_genre', values='rating_count_tot', aggfunc='mean').sort_values('rating_count_tot', ascending=False)

,rating_count_tot
prime_genre,
Navigation,86090.333333
Reference,79350.470588
Social Networking,72916.548077
Music,58205.030769
Weather,52279.892857
Book,46384.916667
Food & Drink,33333.923077
Finance,32367.028571
Travel,29721.605263


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [953]:
ios.query("prime_genre == 'Navigation'")[['track_name', 'rating_count_tot']]

,track_name,rating_count_tot
49,"Waze - GPS Navigation, Maps & Real-time Traffic",345046
130,Google Maps - Navigation & Transit,154911
881,Geocaching®,12811
1633,CoPilot GPS – Car Navigation & Offline Maps,3582
3987,ImmobilienScout24: Real Estate Search in Germany,187
6033,Railway Route Search,5


The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc.

In [954]:
ios.query("prime_genre == 'Social Networking'")[['track_name', 'rating_count_tot']]

,track_name,rating_count_tot
0,Facebook,2974676
5,Pinterest,1061624
43,Skype for iPhone,373519
48,Messenger,351466
51,Tumblr,334293
...,...,...
6380,BestieBox,0
6848,MATCH ON LINE chat,0
6878,niconico ch,0
7090,LINE BLOG,0


Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

In [955]:
ios.query("prime_genre == 'Music'")[['track_name', 'rating_count_tot']]

,track_name,rating_count_tot
4,Pandora - Music & Radio,1126879
8,Spotify Music,878563
35,"Shazam - Discover music, artists, videos & lyrics",402925
60,iHeartRadio – Free Music & Radio Stations,293228
151,SoundCloud - Music & Audio,135744
...,...,...
5096,"Free Music - Player & Streamer for Dropbox, ...",46
5224,NRJ Radio,38
5620,Smart Music: Streaming Videos and Radio,17
5727,BOSS Tuner,13


Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 79,350 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [956]:
ios.query("prime_genre == 'Reference'")[['track_name', 'rating_count_tot']]

,track_name,rating_count_tot
6,Bible,985920
90,Dictionary.com Dictionary & Thesaurus,200047
335,Dictionary.com Dictionary & Thesaurus for iPad,54175
551,Google Translate,26786
715,"Muslim Pro: Ramadan 2017 Prayer Times, Azan, Q...",18418
738,New Furniture Mods - Pocket Wiki & Game Tools ...,17588
757,Merriam-Webster Dictionary,16849
913,Night Sky,12122
1106,City Maps for Minecraft PE - The Best Maps for...,8535
1451,LUCKY BLOCK MOD ™ for Minecraft PC Edition - T...,4693


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

- Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.
    
- Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.
    
- Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.
    

Now let's analyze the Google Play market a bit.

### Most Popular Apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [972]:
android.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver,Installs_Modified
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up,10000
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up,5000000
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up,50000000
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up,100000
5,Paper flowers instructions,ART_AND_DESIGN,4.4,167,5.6M,"50,000+",Free,0,Everyone,Art & Design,"March 26, 2017",1.0,2.3 and up,50000


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

In [958]:
android['Installs_Modified'] = android['Installs'].str.replace('+', '').str.replace(',', '').astype(int)

In [960]:
android.pivot_table(
    index='Category', 
    values='Installs_Modified', 
    aggfunc=lambda x: round(x.mean(), 2)) \
    .sort_values('Installs_Modified', ascending=False)

,Installs_Modified
Category,
COMMUNICATION,38456119.17
VIDEO_PLAYERS,24573948.25
SOCIAL,23253652.13
PHOTOGRAPHY,17805627.64
PRODUCTIVITY,16738957.55
GAME,15578985.37
TRAVEL_AND_LOCAL,13984077.71
ENTERTAINMENT,11640705.88
TOOLS,10696575.50


On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (Skype, Facebook Messenger, Hangouts, WhatsApp, Google Chrome and Gmail), and a few others with over 100 and 500 million installs:

In [976]:
android.query("Category == 'COMMUNICATION'").sort_values('Installs_Modified', ascending=False)[['App', 'Installs']].head(10)

,App,Installs
391,Skype - free IM & video calls,"1,000,000,000+"
382,Messenger – Text and Video Chat for Free,"1,000,000,000+"
464,Hangouts,"1,000,000,000+"
336,WhatsApp Messenger,"1,000,000,000+"
411,Google Chrome: Fast & Secure,"1,000,000,000+"
451,Gmail,"1,000,000,000+"
403,LINE: Free Calls & Messages,"500,000,000+"
420,UC Browser - Fast Download Private & Secure,"500,000,000+"
4676,Viber Messenger,"500,000,000+"
371,Google Duo - High Quality Video Calls,"500,000,000+"


If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [989]:
android \
    .query("Category == 'COMMUNICATION' and Installs_Modified < 100_000_000") \
    .pivot_table(
        index='Category', 
        values='Installs_Modified', 
        aggfunc=lambda x: round(x.mean(), 2)) \
        .sort_values('Installs_Modified', ascending=False)

,Installs_Modified
Category,
COMMUNICATION,3603485.39


In [987]:
android \
    .query("Installs_Modified < 100_000_000") \
    .pivot_table(
        index='Category', 
        values='Installs_Modified', 
        aggfunc=lambda x: round(x.mean(), 2)) \
        .sort_values('Installs_Modified', ascending=False)

,Installs_Modified
Category,
PHOTOGRAPHY,7633342.21
GAME,6216232.10
ENTERTAINMENT,6118250.00
VIDEO_PLAYERS,5508819.34
WEATHER,5074486.20
SHOPPING,4640920.54
COMMUNICATION,3603485.39
PRODUCTIVITY,3369380.60
TOOLS,3195886.77


We see the same pattern for the video players category, which is the runner-up with 24,573,948 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player.

In [990]:
android.query("Category == 'VIDEO_PLAYERS'").sort_values('Installs_Modified', ascending=False)[['App', 'Installs']].head(10)

,App,Installs
3665,YouTube,"1,000,000,000+"
3687,Google Play Movies & TV,"1,000,000,000+"
3711,MX Player,"500,000,000+"
3675,VLC for Android,"100,000,000+"
4032,Dubsmash,"100,000,000+"
4688,VivaVideo - Video Editor & Photo Movie,"100,000,000+"
4696,"VideoShow-Video Editor, Video Maker, Beauty Ca...","100,000,000+"
3672,Motorola Gallery,"100,000,000+"
10647,Motorola FM Radio,"100,000,000+"
3693,LIKE – Magic Video Maker & Community,"50,000,000+"


The pattern is repeated for social apps (where we have giants like Facebook, Google+, Instagram, etc.)

In [992]:
android.query("Category == 'SOCIAL'").sort_values('Installs_Modified', ascending=False)[['App', 'Installs']].head(20)

,App,Installs
2544,Facebook,"1,000,000,000+"
2554,Google+,"1,000,000,000+"
2604,Instagram,"1,000,000,000+"
2546,Facebook Lite,"500,000,000+"
2610,Snapchat,"500,000,000+"
2552,Pinterest,"100,000,000+"
2592,Tango - Live Video Broadcast,"100,000,000+"
2588,Badoo - Free Chat & Dating App,"100,000,000+"
3945,Tik Tok - including musical.ly,"100,000,000+"
3951,BIGO LIVE - Live Stream,"100,000,000+"



or productivity apps (Google Drive, Microsoft Word, Dropbox, Cloud Print, Google Calendar, etc.).

In [994]:
android.query("Category == 'PRODUCTIVITY'").sort_values('Installs_Modified', ascending=False)[['App', 'Installs']].head(10)

,App,Installs
3523,Google Drive,"1,000,000,000+"
3450,Microsoft Word,"500,000,000+"
3473,Dropbox,"500,000,000+"
3574,Cloud Print,"500,000,000+"
3562,Google Calendar,"500,000,000+"
3469,ES File Explorer File Manager,"100,000,000+"
3535,ColorNote Notepad Notes,"100,000,000+"
3493,SwiftKey Keyboard,"100,000,000+"
3489,Samsung Notes,"100,000,000+"
3477,Google Docs,"100,000,000+"


Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,721,912. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [996]:
android.query("Category == 'BOOKS_AND_REFERENCE'").sort_values('Installs_Modified', ascending=False)[['App', 'Installs']].head(20)

,App,Installs
152,Google Play Books,"1,000,000,000+"
3941,Bible,"100,000,000+"
4083,Amazon Kindle,"100,000,000+"
4715,Wattpad 📖 Free Books,"100,000,000+"
5651,Audiobooks from Audible,"100,000,000+"
149,FBReader: Favorite Book Reader,"10,000,000+"
144,Cool Reader,"10,000,000+"
179,Moon+ Reader,"10,000,000+"
173,HTC Help,"10,000,000+"
4100,Aldiko Book Reader,"10,000,000+"


The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average.

However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [1001]:
android \
    .query("1_000_000 < Installs_Modified and Installs_Modified < 100_000_000") \
    .sort_values('Installs_Modified', ascending=False) \
    [['App', 'Installs']]

,App,Installs
4949,"CM Browser - Ad Blocker , Fast Download , Privacy","50,000,000+"
4886,Angry Birds Go!,"50,000,000+"
4874,Angry Birds Friends,"50,000,000+"
10429,Talking Tom Bubble Shooter,"50,000,000+"
10408,Shoot Hunter-Gun Killer,"50,000,000+"
...,...,...
5069,A Better Camera,"5,000,000+"
5063,Blur Image - DSLR focus effect,"5,000,000+"
5048,AppFinder by AppTap,"5,000,000+"
5033,ActionDirector Video Editor - Edit Videos Fast,"5,000,000+"


This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

## Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

Here are a few next steps you could take:

- Analyze the frequency table for the Genre column of the Google Play dataset, and see if you can find useful patterns.
- Assume we could also make revenue via in-app purchases and subscriptions, and try to determine which genres seem to be liked the most by users — you could examine app ratings here.
- Refine your project using our data science project [style guide](https://www.dataquest.io/blog/data-science-project-style-guide/).